lang/src/org/partiql/lang/eval/physical/operators/DefaultOperatorFactories.kt

```diff
@@ -24,6 +26,51 @@ import org.partiql.lang.planner.transforms.DEFAULT_IMPL_NAME
  * @see [org.partiql.lang.planner.PlannerPipeline.Builder.addRelationalOperatorFactory]
  */
 internal val DEFAULT_RELATIONAL_OPERATOR_FACTORIES = listOf(
+
+    object : UnpivotRelationalOperatorFactory(DEFAULT_IMPL_NAME) {
+        override fun create(
+            impl: PartiqlPhysical.Impl,
+            expr: ValueExpression,
+            setAsVar: SetVariableFunc,
+            setAtVar: SetVariableFunc?,
+            setByVar: SetVariableFunc?
+        ): RelationExpression =
+            RelationExpression { state ->
+
+                fun ExprValue.unpivot(state: EvaluatorState): ExprValue = when (type) {
+                    ExprValueType.STRUCT, ExprValueType.MISSING -> this
+
+                    else -> state.valueFactory.newBag(
+                        listOf(
+                            this.namedValue(state.valueFactory.newString(syntheticColumnName(0)))
+                        )
+                    )
+                }
+
+                val originalValue = expr(state)
+
+                val unpivot = originalValue.unpivot(state)
+
+                relation(RelationType.BAG) {
+                    val iter = unpivot.iterator()
+                    while (iter.hasNext()) {
+                        val item = iter.next()
+
+                        setAsVar(state, item.unnamedValue())
+
+                        if (setAtVar != null) {
+                            setAtVar(state, item.name ?: state.valueFactory.missingValue)
+                        }
+
+                        if (setByVar != null) {
+                            setByVar(state, item.address ?: state.valueFactory.missingValue)
+                        }
+                        yield()
+                    }
+                }
+            }
+    },
```

Comment:
As John did with ORDER BY, let's consider defining our operators as standalone classes. It's a bit odd to define our operators as anonymous objects in a global list literal. If an operator is its own class, the we can unit test it. As is, we don't have any unit tests for individual operators (only end-to-end evaluation tests). Because these are internals, we can refactor later, but for now let's define new operators as their own classes and shim appropriately with what we are working with.

Reply:
Agree. I can create an issue for that and refactor for all the existing operator.